src/common/storage/util.js

```diff
@@ -56,6 +199,76 @@ define(['common/storage/constants', 'common/util/jsonPatcher'], function (CONSTA
             return !!(coreObj.oldHash && coreObj.newHash && coreObj.oldData && coreObj.newData);
         },
         getChangedNodes: jsonPatcher.getChangedNodes,
-        applyPatch: jsonPatcher.apply
+        applyPatch: jsonPatcher.apply,
+
+        /**
+         * Extracts a serializable json representation of a project tree.
+         * To specify starting point set one of the four options. If more than one is set the order of precedence is:
+         * rootHash, commitHash, tagName and branchName.
```

Comment:
You mentioned the different order of look up and implemented it too - please update it here too.